src/status_im/contexts/wallet/events.cljs

```diff
@@ -77,7 +77,11 @@
 
 (rf/reg-event-fx :wallet/switch-current-viewing-account
  (fn [{:keys [db]} [address]]
-   {:db (assoc-in db [:wallet :current-viewing-account-address] address)}))
+   (let [{:keys [tx-type]} (db/send db)
+         bridge-tx?        (= tx-type :tx/bridge)]
+     {:db (cond-> db
+            :always    (assoc-in [:wallet :current-viewing-account-address] address)
+            bridge-tx? (update-in db-path/send assoc :to-address address))})))
```

Comment:
ideally, instead of adding send specific updates here, we could have to-address always be the same as the current-viewing-address.

Reply:
I have some topics to discuss in DX meeting, and if all good we can improve logic in a refactoring PR.